diff --git a/include/boost/format/alt_sstream_impl.hpp b/include/boost/format/alt_sstream_impl.hpp
index 998f8b2..3872ee2 100644
--- a/include/boost/format/alt_sstream_impl.hpp
+++ b/include/boost/format/alt_sstream_impl.hpp
@@ -38,10 +38,10 @@ namespace boost {
             size_type sz=s.size();
             if(sz != 0 && mode_ & (::std::ios_base::in | ::std::ios_base::out) ) {
 #ifdef _RWSTD_NO_CLASS_PARTIAL_SPEC
-                void *vd_ptr = alloc_.allocate(sz, is_allocated_? eback() : 0);
+                void *vd_ptr = alloc_.allocate(sz);
                 Ch *new_ptr = static_cast<Ch *>(vd_ptr);
 #else
-                Ch *new_ptr = alloc_.allocate(sz, is_allocated_? eback() : 0);
+                Ch *new_ptr = alloc_.allocate(sz);
 #endif
                 // if this didnt throw, we're safe, update the buffer
                 dealloc();
@@ -255,10 +255,10 @@ namespace boost {
                 if(0 < add_size) {
                     new_size += add_size;
 #ifdef _RWSTD_NO_CLASS_PARTIAL_SPEC
-                    void *vdptr = alloc_.allocate(new_size, is_allocated_? oldptr : 0);
+                    void *vdptr = alloc_.allocate(new_size);
                     newptr = static_cast<Ch *>(vdptr);
 #else
-                    newptr = alloc_.allocate(new_size, is_allocated_? oldptr : 0);
+                    newptr = alloc_.allocate(new_size);
 #endif
                 }
 
